pjmedia/src/pjmedia/conf_thread.c

```diff
@@ -0,0 +1,3858 @@
+            buff_ptime = port_ptime;
+            if (port_ptime % conf_ptime)
+                buff_ptime += conf_ptime;
+        } else {
+            buff_ptime = conf_ptime;
+            if (conf_ptime % port_ptime)
+                buff_ptime += port_ptime;
+        }
+
+        /* Create RX buffer. */
+        //conf_port->rx_buf_cap = (unsigned)(conf_port->samples_per_frame +
+        //                                 conf->samples_per_frame * 
+        //                                 conf_port->clock_rate * 1.0 /
+        //                                 conf->clock_rate + 0.5);
+        conf_port->rx_buf_cap = conf_port->sampling_rate * buff_ptime / 1000;
+        if (conf_port->channel_count > conf->channel_count)
+            conf_port->rx_buf_cap *= conf_port->channel_count;
+        else
+            conf_port->rx_buf_cap *= conf->channel_count;
+
+        conf_port->rx_buf_count = 0;
+        conf_port->rx_buf = (pj_int16_t*)
+                            pj_pool_alloc(pool, conf_port->rx_buf_cap *
+                                                sizeof(conf_port->rx_buf[0]));
+        PJ_ASSERT_ON_FAIL(conf_port->rx_buf,
+                          {status = PJ_ENOMEM; goto on_return;});
+
+        /* Create TX buffer. */
+        conf_port->tx_buf_cap = conf_port->rx_buf_cap;
+        conf_port->tx_buf_count = 0;
+        conf_port->tx_buf = (pj_int16_t*)
+                            pj_pool_alloc(pool, conf_port->tx_buf_cap *
+                                                sizeof(conf_port->tx_buf[0]));
+        PJ_ASSERT_ON_FAIL(conf_port->tx_buf,
+                          {status = PJ_ENOMEM; goto on_return;});
+    }
+
+
+    /* Create mix buffer. */
+    conf_port->mix_buf = (pj_int32_t*)
+                         pj_pool_zalloc(pool, conf->samples_per_frame *
+                                              sizeof(conf_port->mix_buf[0]));
+    PJ_ASSERT_ON_FAIL(conf_port->mix_buf,
+                      {status = PJ_ENOMEM; goto on_return;});
+    conf_port->last_mix_adj = NORMAL_LEVEL;
+
+    /* init auto adjustment level for mixed signal. */
+    conf_port->mix_adj = NORMAL_LEVEL;
+
+    /* We need not reset mix_buf, we just want to copy the first
+    * (and probably only) frame there.
+    * The criteria for "this frame is from the first transmitter"
+    *  condition is:
+    * (conf_port->last_timestamp.u64 != frame->timestamp.u64)
+    * 
+    * Enforce "this frame is from the first transmitter" condition:
+    * initially we set last_timestamp to -1, so the first frame will be copied.
+    */
+    conf_port->last_timestamp.u64 = (pj_uint64_t)-1;
+
+    if (conf->is_parallel) {
+        PJ_TEST_NOT_NULL(conf_port->rx_frame_buf = (pj_int16_t*)pj_pool_zalloc(pool, conf->rx_frame_buf_cap), NULL, 
+                         {status = PJ_ENOMEM;goto on_return;});
+        PJ_TEST_SUCCESS(status=pj_lock_create_simple_mutex(pool, "tx_lock", &conf_port->tx_lock), NULL, goto on_return);
```

Comment:
Although it offers the desired functionality here, I believe unittest.h, and PJ_TEST* is not supposed to be used other than for testing purposes.

If you need it, I would suggest to define your own macro in this file.